### PR TITLE
Add Tutorials `6. Store Encrypted Data` and `7. Fetch Encrypted Data`

### PR DESCRIPTION
**Pull request by lzpap**
## Description
- Add `6. Store Encrypted Data` tutorial to show how one can put encrypted data on the Tangle. 
- Add `7. Fetch Encrypted Data` tutorial to show how one can fetch and read encrypted data from the Tangle
- The two tutorials need to be studied one after the other.

### CODE REVIEW

File: docs/tutorials.rst [hunk 1]
```diff
@@ -505,9 +505,224 @@ Once the bundle is confirmed, try rerunning the script from
...
+~~~~~~~~~~
+.. literalinclude:: ../examples/tutorials/06_store_encrypted.py
+   :lines: 1-15
+   :lineno-start: 1
+
+The first odd thing we notice among the imports is
+:py:class:`~iota.crypto.addresses.AddressGenerator`. It is a PyOTA specific
+utility that generates addresses given your seed.
+
+We will use the ``encrypt`` method to enchiper the data, and ``b64encode`` for
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
We will use the ``encrypt`` method to encipher the data, and ``b64encode`` for
```

File: docs/tutorials.rst [hunk 2]
```diff
@@ -505,9 +505,224 @@ Once the bundle is confirmed, try rerunning the script from
...
+Therefore, we first encode our binary data into ASCII characters with `Base64`_
+encoding.
+
+.. literalinclude:: ../examples/tutorials/06_store_encrypted.py
+   :lines: 39-57
+   :lineno-start: 39
+
+Now, we are ready to construct the transfer. We convert the encrypted `Base64`_
+encoded data to trytes and assign it to the :py:class:`ProposedTransaction`
+object's message argument.
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
object's ``message`` argument.
```

File: docs/tutorials.rst [hunk 3]
```diff
@@ -505,9 +505,224 @@ Once the bundle is confirmed, try rerunning the script from
...
+
+An address is also needed, so we generate one with the help of
+:py:class:`~iota.crypto.addresses.AddressGenerator` and its
+:py:meth:`~iota.crypto.addresses.AddressGenerator.get_addresses` method. Feel
+free to chose the index of the generated address, and don't forget, that the
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
free to choose the index of the generated address, and don't forget, that the
```

File: docs/tutorials.rst [hunk 4]
```diff
@@ -505,9 +505,224 @@ Once the bundle is confirmed, try rerunning the script from
...
+
+.. literalinclude:: ../examples/tutorials/07_fetch_encrypted.py
+   :lines: 16-17
+   :lineno-start: 16
+
+To fetch transactions or bundles from the Tangle, a reference is required to
+retreive them from the network. Transactions are identified by their
+transaction hash, while a group of transaction (a bundle) by bundle hash.
+Hashes ensure the integrity of the Tangle, since they contain verifiable
+information on the content of the transfer objects.
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
information about the content of the transfer objects.
```

File: docs/tutorials.rst [hunk 5]
```diff
@@ -505,9 +505,224 @@ Once the bundle is confirmed, try rerunning the script from
...
+transaction hashes and returns the bundles for each of them. The response
+``dict`` contains a ``bundles`` key with the value being a list of bundles
+in the same order as the input argument hashes. Also note, that the bundles
+in the response are actual PyOTA :py:class:`Bundle` objects.
+
+To simplify the code, several operations are happening on line 21:
+
+- Calling :py:meth:`~Iota.get_bundles` that returns a ``dict``,
+- accessing the ``'bundles'`` key in the ``dict``,
+- and taking the first element of the the list if bundles in the value
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
- and taking the first element of the the list of bundles in the value
```

File: docs/tutorials.rst [hunk 6]
```diff
@@ -505,9 +505,224 @@ Once the bundle is confirmed, try rerunning the script from
...
+   :lineno-start: 23
+
+The next step is to extract the content of the message fields of the
+transactions in the bundle. We call :py:meth:`Bundle.get_messages` to carry
+out this operation. The method returns a list of unicode strings, essentially
+the ``signature_message_fragment`` fields of the transactions, decoded from
+trytes into unicode characters.
+
+We then combine these message chunks into one stream of characters by using
+``sting.join()``.
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
``string.join()``.
```

File: docs/tutorials.rst [hunk 7]
```diff
@@ -505,9 +505,224 @@ Once the bundle is confirmed, try rerunning the script from
...
+
+We know that at this stage that we can't make sense of our message, because it
+is encrypted and encoded into `Base64`_. Let's peel that onion layer by layer:
+
+- On line 28, we decode the message into bytes with ``b64decode``.
+- On line 31, we ask the user for a decryption password.
```
Comment (todofixthis) [inline docs/tutorials.rst]:
```suggestion
- On line 31, we ask the user for the decryption password (from the previous tutorial).
```

File: examples/tutorials/06_store_encrypted.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+b64_cipher = b64encode(cipher)
+
+print('Constructing transaction locally...')
+# Convert to trytes
+trytes_encrypted_data = TryteString.from_bytes(b64_cipher)
+
+# Generate an address from your seed to post the transfer to
+my_address = AddressGenerator(b'YOURSEEDFROMTHEPREVIOUSTUTORIAL').get_addresses(
+    start=42,
+)[0]
```
Comment (todofixthis) [inline examples/tutorials/06_store_encrypted.py]: ❔ Could we provide the seed to `Iota.__init__()` on line 15, and then use `api.get_new_addresses()` here instead?

Comment (lzpap) [reply examples/tutorials/06_store_encrypted.py]: Absolutely! It's less confusing as well, there is no need for user to know about the existence of `AddressGenerator`. :)

**Pull request merged**